lib/matplotlib/tests/test_text.py

```diff
@@ -279,8 +279,127 @@ def test_get_rotation_mod360():
     for i, j in zip([360., 377., 720+177.2], [0., 17., 177.2]):
         assert_almost_equal(text.get_rotation(i), j)
 
+
 @image_comparison(baseline_images=['text_bboxclip'])
 def test_bbox_clipping():
     plt.text(0.9, 0.2, 'Is bbox clipped?', backgroundcolor='r', clip_on=True)
     t = plt.text(0.9, 0.5, 'Is fancy bbox clipped?', clip_on=True)
     t.set_bbox({"boxstyle": "round, pad=0.1"})
+
+
+def test_text_annotation_get_window_extent():
```

Comment:
It is easier to add the `@cleanup` decorator use pyplot to make your figure/axes objects.